cmd/init_project.go

```diff
@@ -185,30 +191,54 @@ func boilerplateForFlags(f *flag.FlagSet) *project.Boilerplate {
 	f.StringVar(&b.Owner, "owner", "", "Owner to add to the copyright")
 	return b
 }
-func checkGoVersion() {
+
+func ensureGoVersionIsCompatible() {
+	err := fetchAndCheckGoVersion()
+	if err != nil {
+		log.Fatalf("%s. You can skip this check using the --skip-go-version-check flag", err)
+	}
+}
+
+func fetchAndCheckGoVersion() error {
 	cmd := exec.Command("go", "version")
 	out, err := cmd.Output()
 	if err != nil {
-		log.Fatalf("Could not execute 'go version': %v", err)
+		return fmt.Errorf("Failed to retrieve 'go version': %v", string(out))
 	}
 
 	split := strings.Split(string(out), " ")
 	if len(split) < 3 {
-		log.Fatalf("Invalid go version: %q", string(out))
+		return fmt.Errorf("Found invalid Go version: %q", string(out))
 	}
-	goVersion := strings.TrimPrefix(split[2], "go")
-	if ver, err := semver.NewVersion(goVersion); err != nil {
-		if err != nil {
-			log.Fatalf("Invalid go version %q: %v", goVersion, err)
-		}
-		c, err := semver.NewConstraint(">= 1.10")
-		if err != nil {
-			log.Fatal("Invalid constraint: %v", err)
-		}
-		if !c.Check(ver) {
-			log.Fatalf("The go version is %v, must be 1.10+", goVersion)
-		}
+	goVer := split[2]
+	if err := checkGoVersion(goVer); err != nil {
+		return fmt.Errorf("Go version '%s' is incompatible because '%s'", goVer, err)
+	}
+	return nil
+}
+
+func checkGoVersion(verStr string) error {
+	goVerRegex := `^go?([0-9]+)\.([0-9]+)([\.0-9A-Za-z\-]+)?$`
+	m := regexp.MustCompile(goVerRegex).FindStringSubmatch(verStr)
+	if m == nil {
+		return fmt.Errorf("invalid version string")
+	}
+
+	major, err := strconv.ParseInt(m[1], 10, 64)
```

Comment:
nit: I guess using `strcov.Atoi` is OK and easier to understand.

Reply:
Done.